src/main.rs

```diff
@@ -266,26 +266,33 @@ fn configure_app(app: App<MiniserveConfig>) -> App<MiniserveConfig> {
             })
             // Handle directories
             .handler(&full_route, s)
+            .default_resource(|r| r.method(Method::GET).f(p404))
         } else {
             // Handle directories
             app.handler(&full_route, s)
+                .default_resource(|r| r.method(Method::GET).f(p404))
         }
     } else {
         // Handle single files
         app.resource(&full_route, |r| r.f(listing::file_handler))
+            .default_resource(|r| r.method(Method::GET).f(p404))
     }
 }
 
 fn p404(req: &HttpRequest<crate::MiniserveConfig>) -> Result<HttpResponse, io::Error> {
```

Comment:
What does the `p` stand for?

Reply:
I guess it means page, I took it from actix examples directly. Do you want me to rename it?

Reply:
Yes, as you probably already figured, I like more explicit names generally.

Reply:
How about `error_404`?

Reply:
Sure ;)